packages/ui/Scrollable/Scrollable.js

```diff
@@ -64,710 +64,499 @@ const TouchableDiv = ForwardRef({prop: 'ref'}, Touchable('div'));
+				bounds = getScrollBounds(),
+				bCanScrollHorizontally = canScrollHorizontally(bounds),
+				bCanScrollVertically = canScrollVertically(bounds),
 				eventDeltaMode = ev.deltaMode,
 				eventDelta = (-ev.wheelDeltaY || ev.deltaY);
 			let
 				delta = 0,
 				direction;
 
-			this.lastInputType = 'wheel';
+			variables.current.lastInputType = 'wheel';
 
-			if (this.props.noScrollByWheel) {
+			if (props.noScrollByWheel) {
 				return;
 			}
 
-			if (canScrollVertically) {
-				delta = this.calculateDistanceByWheel(eventDeltaMode, eventDelta, bounds.clientHeight * scrollWheelPageMultiplierForMaxPixel);
-			} else if (canScrollHorizontally) {
-				delta = this.calculateDistanceByWheel(eventDeltaMode, eventDelta, bounds.clientWidth * scrollWheelPageMultiplierForMaxPixel);
+			if (bCanScrollVertically) {
+				delta = calculateDistanceByWheel(eventDeltaMode, eventDelta, bounds.clientHeight * scrollWheelPageMultiplierForMaxPixel);
+			} else if (bCanScrollHorizontally) {
+				delta = calculateDistanceByWheel(eventDeltaMode, eventDelta, bounds.clientWidth * scrollWheelPageMultiplierForMaxPixel);
 			}
 
 			direction = Math.sign(delta);
 
-			if (direction !== this.wheelDirection) {
-				this.isScrollAnimationTargetAccumulated = false;
-				this.wheelDirection = direction;
+			if (direction !== variables.current.wheelDirection) {
+				variables.current.isScrollAnimationTargetAccumulated = false;
+				variables.current.wheelDirection = direction;
 			}
 
-			forward('onWheel', {delta, horizontalScrollbarRef, verticalScrollbarRef}, this.props);
+			forward('onWheel', {delta, horizontalScrollbarRef, verticalScrollbarRef}, props);
 
 			if (delta !== 0) {
-				this.scrollToAccumulatedTarget(delta, canScrollVertically, this.props.overscrollEffectOn.wheel);
+				scrollToAccumulatedTarget(delta, bCanScrollVertically, props.overscrollEffectOn.wheel);
 				ev.preventDefault();
 				ev.stopPropagation();
 			}
 		}
-	}
+	});
 
-	scrollByPage = (keyCode) => {
+	function scrollByPage (keyCode) {
 		const
-			bounds = this.getScrollBounds(),
-			canScrollVertically = this.canScrollVertically(bounds),
-			pageDistance = (isPageUp(keyCode) ? -1 : 1) * (canScrollVertically ? bounds.clientHeight : bounds.clientWidth) * paginationPageMultiplier;
+			bounds = getScrollBounds(),
+			bCanScrollVertically = canScrollVertically(bounds),
+			pageDistance = (isPageUp(keyCode) ? -1 : 1) * (bCanScrollVertically ? bounds.clientHeight : bounds.clientWidth) * paginationPageMultiplier;
 
-		this.lastInputType = 'pageKey';
+		variables.current.lastInputType = 'pageKey';
 
-		this.scrollToAccumulatedTarget(pageDistance, canScrollVertically, this.props.overscrollEffectOn.pageKey);
+		scrollToAccumulatedTarget(pageDistance, bCanScrollVertically, props.overscrollEffectOn.pageKey);
 	}
 
-	onKeyDown = (ev) => {
-		if (this.props.onKeyDown) {
-			forward('onKeyDown', ev, this.props);
+	const onKeyDown = useCallback((ev) => {
```

Comment:
No dependency means `props.onKeyDown` will not be changed ever since the callback function is memoized. I'm wondering this is intended.

Reply:
As far as I know, No dependency means `props.onKeyDown` will be changed everytimes. it is same with normal function. It is meanless useCallback.

I hope to fill the dep parameter properly. But, I need to set all functions that called inside.
and the inside function should be useCallback function too.

and there are many connections in the dep connection.
I hope to solve this part onrestructuring with `// TODO: Handle exhaustive-deps`